### PR TITLE
Extra packets

### LINKED ISSUE #70: Implement network packets needed for round manager
Packets to be implemented are:

- [x] Round Start: Server -> Client
- [x] Pre Round Start: Server -> Client
- [x] Round End: Server -> Client
- [x] Space: Client -> Server (in response to Round End)
- [x] Eliminate Players: Server -> Client

Data to be transferred each packet currently is as follows:
**Pre Round Start**:
- round number(byte)
- pre round length(byte, should only be up to 255s)
- round length(byte, hopefully a round isn’t longer than 255 secs)
- number of players(byte), 
- spaces activated(list of bytes).

**Round Start**:
- round number(byte).

**Round End**:
- round number(byte).

**Space**: 
- round number(byte)
- spaceID(byte, im assuming no more than 255 unique spaces on the map, including the ones that aren’t active).

**Eliminate Players**:
- round number(byte)
- players eliminated (list of player ids). Whatever data type we use to store player IDs in the world/server. I can’t remember what that is.

_Originally posted by @RileyEv in https://github.com/BigMacintosh/Parking/issues/64#issuecomment-579312981_

### PR DESCRIPTION
**Pull request by liamdalg**
Implements all of Riley's requested round packets. Resolves #70

### CODE REVIEW

File: ProjectFiles/Assets/Scripts/Network/Events/ServerEliminatePlayersEvent.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                writer.Write(p);
+            }
+        }
+
+        public override void Deserialise(DataStreamReader reader, ref DataStreamReader.Context context)
+        {
+            RoundNumber = reader.ReadByte(ref context);
+
+            var playerCount = reader.ReadByte(ref context);
+            Players = new List<byte>();
```
Comment (ghost) [inline ProjectFiles/Assets/Scripts/Network/Events/ServerEliminatePlayersEvent.cs]: should the list maybe be initialised in constructor?

Comment (liamdalg) [reply ProjectFiles/Assets/Scripts/Network/Events/ServerEliminatePlayersEvent.cs]: Yeah, I think that's what we did elsewhere. Although I was considering in the future we might want to switch to arrays over lists (when efficiency becomes important)? Can switch for now though

File: ProjectFiles/Assets/Scripts/Network/Events/ServerPreRoundStartEvent.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        public byte RoundNumber { get; private set; }
+        public byte PreRoundLength { get; private set; }
+        public byte RoundLength { get; private set; }
+        public byte PlayerCount { get; private set; }
+        public List<byte> Spaces { get; private set; }
+
+        public ServerPreRoundStartEvent()
+        {
+            ID = EventType.ServerPreRoundStartEvent;
+            Length = 1;
```
Comment (RileyEv) [inline ProjectFiles/Assets/Scripts/Network/Events/ServerPreRoundStartEvent.cs:17]: This is a bit weird

**Review by RileyEv — COMMENTED** [review]
It doesn't really make sense to have a length of 1 for a packet that can have a variable length. Should have a predefined constant? I'm not sure 🤔

**Review by ghost — APPROVED** [review]

**Comment by ghost** [comment]
very cool, great job Liam :)

**Review by RileyEv — APPROVED** [review]

**Pull request merged**